From 4198f00d52a9bb9c9f7ad7e5dcc2d6ba16a1992c Mon Sep 17 00:00:00 2001
From: Valentine Barshak <valentine.barshak@cogentembedded.com>
Date: Mon, 17 May 2021 17:18:50 +0300
Subject: [PATCH 66/70] staging: most: dim2: Add device_release callback

This adds device_release callback to the dim2 most device interface
which fixes the following kernel error:

  "Device 'dim2-00000000ec520000' does not have a release()
   function, it is broken and must be fixed.
   See Documentation/core-api/kobject.rst."

Signed-off-by: Valentine Barshak <valentine.barshak@cogentembedded.com>
---
 drivers/staging/most/dim2/dim2.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/staging/most/dim2/dim2.c b/drivers/staging/most/dim2/dim2.c
index 6127f27..4f0c096 100644
--- a/drivers/staging/most/dim2/dim2.c
+++ b/drivers/staging/most/dim2/dim2.c
@@ -675,6 +675,11 @@ static void dma_free(struct mbo *mbo, u32 size)
 	dma_free_coherent(dev, size, mbo->virt_address, mbo->bus_address);
 }
 
+static void dim2_dev_release(struct device *dev)
+{
+	/* dummy release */
+}
+
 static const struct of_device_id dim2_of_match[];
 
 static struct {
@@ -856,6 +861,7 @@ static int dim2_probe(struct platform_device *pdev)
 	dev->most_iface.dev = &dev->dev;
 	dev->dev.init_name = dev->name;
 	dev->dev.parent = &pdev->dev;
+	dev->dev.release = dim2_dev_release;
 
 	ret = most_register_interface(&dev->most_iface);
 	if (ret) {
-- 
2.7.4

